src/lib/featureFlags/featureFlags.ts

```diff
@@ -0,0 +1,30 @@
+import { FeatureFlagConfig, FeatureFlagProvider } from "./featureFlagProvider"
+
+const registeredProviders = new Map<symbol, any>()
+
+export async function createFlagProvider<T extends FeatureFlagConfig>(
+  type: symbol,
+  config?: T
+): Promise<FeatureFlagProvider> {
+  if (!registeredProviders.has(type)) {
```

Comment:
ditto here. all of the variable names should be consistent if possible. There's now `FeatureFlagProvider`, flagProvider`, registeredProvider`, `registerFlagProvider`. We should stick with one idiom throughout